src/Documentation/grains/transactions.md

```diff
@@ -88,31 +116,54 @@ public class AccountGrain : Grain, IAccountGrain
     private readonly ITransactionalState<Balance> balance;
 
     public AccountGrain(
-        [TransactionalState("balance")] ITransactionalState<Balance> balance)
+        [TransactionalState("balance", "TransactionStore")]
+        ITransactionalState<Balance> balance)
     {
         this.balance = balance ?? throw new ArgumentNullException(nameof(balance));
     }
 
     Task IAccountGrain.Deposit(uint ammount)
     {
-        this.balance.State.Value += ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value += amount);
     }
 
     Task IAccountGrain.Withdrawal(uint ammount)
     {
-        this.balance.State.Value -= ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value -= amount);
     }
 
     Task<uint> IAccountGrain.GetBalance()
     {
-        return Task.FromResult(this.balance.State.Value);
+        return this.balance.PerformRead(x => x.Value);
     }
-}
+}`
+```
+
+In the above example, the attribute `TransactionalState` is used to declare that the ‘balance’ constructor argument should be associated with a transactional state named “balance”.
+With this declaration, Orleans will wire up an `ITransactionalState` instance with a state loaded from the transactional state storage named "TransactionStore" (see Setup).
+The state can be modified via `PerformUpdate` or read via `PerformRead`.
+The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted upon completion of the grain call which created the transaction (`IATMGrain.Transfer` in the above examples).
+
+### Calling Transactions
+
+Transactional methods on a grain interface are called the same as any other grain call.
```

Comment:
"the same" -> "the same way"? I'm not sure what sounds more natural

Reply:
I agree. How about this:

>Transactional methods on a grain interface are called like any other grain call.